### ISSUE TITLE
Check how we are getting `-L` and `-l` flags from GASNet

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#23362]
In reviewing https://github.com/chapel-lang/chapel/pull/23321#discussion_r1323191915, @mppf noted that it ought to be possible for us to more precisely sort flags related to GASNet linkage into those that point to our bundled libraries vs. those that GASNet provides to point to system libraries:

>  the ideal way to solve it would be for chpl_gasnet.get_link_args() to properly divide the -L / -l args between bundled and system. In particular, that's supposed to return a 2-tuple of lists; so that the bundled ones end up in CHPL_TARGET_BUNDLED_LINK_ARGS but the system ones end up in CHPL_TARGET_SYSTEM_LINK_ARGS.

This sounds like a good idea, but wasn't one I was able to explore before the release, so I'm filing this issue as a future work item (where that future could be immediately if someone else has cycles to pick it up).

**Comment by mppf** [comment]
Another way to solve this would be to rename all of the bundled 3rd party libraries we build e.g. from libjemalloc.a to libchpl_jemalloc.a.

**Comment by bradcray** [comment]
Even if we were to do that, presumably we should still be sorting the GASNet flags into the right categories as long as that distinction exists in our variables and scripts, right?

**Comment by mppf** [comment]
Yes, but we could perhaps forget about making the distinction & simplify some of this stuff, if we went with the renaming strategy.
